Sources/SwiftkubeClient/Config/KubernetesClientConfig.swift

```diff
@@ -228,3 +228,51 @@ private extension AuthInfo {
 		return nil
 	}
 }
+
+public extension KubernetesClientConfig {
+	static func loadConfigFromPath(logger: Logger, path: String) throws -> KubernetesClientConfig? {
+		let decoder = YAMLDecoder()
+
+		let fileURL = URL(fileURLWithPath: path)
+
+		guard let contents = try? String(contentsOf: fileURL, encoding: .utf8) else {
+			return nil
+		}
+
+		guard let kubeConfig = try? decoder.decode(KubeConfig.self, from: contents) else {
+			return nil
+		}
+
+		guard let currentContext = kubeConfig.currentContext else {
+			return nil
+		}
+
+		guard let context = kubeConfig.contexts?.filter({ $0.name == currentContext }).map(\.context).first else {
+			return nil
+		}
+
+		guard let cluster = kubeConfig.clusters?.filter({ $0.name == context.cluster }).map(\.cluster).first else {
+			return nil
+		}
+
+		guard let masterURL = URL(string: cluster.server) else {
+			return nil
+		}
+
+		guard let authInfo = kubeConfig.users?.filter({ $0.name == context.user }).map(\.authInfo).first else {
+			return nil
+		}
+
+		guard let authentication = authInfo.authentication(logger: logger) else {
+			return nil
+		}
+
+		return KubernetesClientConfig(
+			masterURL: masterURL,
+			namespace: context.namespace ?? "default",
+			authentication: authentication,
+			trustRoots: cluster.trustRoots(logger: logger),
+			insecureSkipTLSVerify: cluster.insecureSkipTLSVerify ?? true
+		)
+	}
```

Comment:
This is an exact duplicate of the same lines in the `LocalFileConfigLoader`. How about this:

- We change the `LocalFileConfigLoader` to accept a `URL` and the body would contains this logic for loading any given arbitrary `URL`.

```swift
internal struct LocalFileConfigLoadera {
   internal func load(fromURL: URL, logger: Logger) throws -> KubernetesClientConfig? { ... }
} 
```
- Then we rename `LocalFileConfigLoader` to something like `URLConfigLoader`
- And finally rename the new `KubernetesClientConfig` to something like `LocalKubeConfigLoader` and make it call the `URLConfigLoader` to restore the previous functionality, e.g.:

```swift
struct LocalKubeConfigLoader {
    func load(logger: Logger) throws -> KubernetesClientConfig? {
        guard let homePath = ProcessInfo.processInfo.environment["HOME"] else {
            logger.info("Skipping kubeconfig in $HOME/.kube/config because HOME env variable is not set.")
            return nil
        }
        let kubeConfigURL = URL(fileURLWithPath: homePath + "/.kube/config")
        return try? URLConfigLoader().load(fromURL: kubeConfigURL, logger: logger)
    }
}
```

Reply:
Ah I see. So we will have three structs the implement `KubernetesClientConfigLoader`
1. LocalFileConfigLoader
2. LocalKubeConfigLoader
3. ServiceAccountConfigLoader
Cool, I reckon that'll work well 😊

Reply:
Yes exactly, however the `LocalFileConfigLoader` can accept any `URL`, even remote URLs` (don't know if that is really a use-case), so we could just call it `URLConfigLoader` 😉